### PR TITLE
Feat/answers on form

### PR DESCRIPTION
**Pull request by alphonseb**
This PR adds an Algolia Answer search when typing the subject of the request on a zendesk contact page, and displays the best answer (if it exists).

Note: with the way this is done, the user would be required to insert in zendesk one code block on the homepage template (for the autocomplete), and one code block on the new request template. To make the integration easier, we could possibly refactor so that only one code block has to be inserted (in the document head template for instance, with of course the downside that it would be loaded on every zendesk page). Happy to discuss it!

### CODE REVIEW

**Comment by redox** [comment]
> Note: with the way this is done, the user would be required to insert in zendesk one code block on the homepage template (for the autocomplete), and one code block on the new request template.

I think this is acceptable :)

File: app/src/ticketForm.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+      const warning = document.createElement('span');
+      warning.classList.add(descriptionWarning);
+      warning.textContent = translate(translations, locale, 'descriptionLock');
+      self.descriptionElement.append(warning);
+    }
+
+    document
+      .querySelector(suggestionsListSelector)
+      .setAttribute('style', 'display: none!important');
+
```
Comment (redox) [inline app/src/ticketForm.js:74]: Same, let's ensure we add some nice error handling is `suggestionsListSelector` is not found.

File: app/src/ticketForm.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    const lang = locale.split('-')[0];
+    const buildUrl = (hit) => `${baseUrl}${locale}/articles/${hit.id}`;
```
Comment (redox) [inline app/src/ticketForm.js:76]: Let's probably move `buildUrl` to a helper/util file, as it's used in 2 files now.

File: app/src/ticketForm.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            }
+          },
+          autocomplete: false,
+        });
+      };
+
+      const onClick = (e, item) => {
+        e.preventDefault();
+        self.trackClick(item, item.__position, item.__queryID);
+        window.location = e.target.href;
```
Comment (redox) [inline app/src/ticketForm.js:118]: I think you can just do 

```js
const onClick = (e, item) => { self.trackClick(item, item.__position, item.__queryID); }
```

and not prevent the default behavior which you seem to anyway reimplement with `
window.location = e.target.href;` isn't it?

File: app/src/ticketForm.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+                              }),
+                            }}
+                          />
+                        </h4>
+                        <p
+                          dangerouslySetInnerHTML={{
+                            __html: snippet({
+                              attribute: 'body_safe',
+                              hit,
+                              highlightedTagName: 'strong',
```
Comment (redox) [inline app/src/ticketForm.js:162]: Use `mark` instead of `strong` for highlighing, better semantically.

File: app/src/ticketForm.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+              </div>
+            </div>
+          ) : null}
+        </Fragment>
+      );
+    };
+
+    render(
+      h(Input),
+      document.querySelector(inputSelector).parentNode,
```
Comment (redox) [inline app/src/ticketForm.js:179]: validate if `inputSelector` is found?

File: app/src/translations.js [hunk 6]
```diff
@@ -549,6 +549,39 @@ const defaultTranslations = {
...
+    ru: 'Пожалуйста, сначала заполните тему.',
+    sk: 'Najskôr vyplňte predmet.',
+    sv: 'Najprej izpolnite predmet.',
+    th: 'กรุณากรอกหัวข้อก่อน',
+    tr: 'Lütfen önce konuyu doldurunuz.',
+    uk: 'Спочатку заповніть тему.',
+    vi: 'Vui lòng điền chủ đề trước.',
+    'zh-cn': '请先填写主题。',
+    'zh-tw': '請先填寫主題。',
+  },
```
Comment (redox) [inline app/src/translations.js:584]: Nice :) I've used GSheet's `GOOGLETRANSLATE` function for this in the past :)

**Review by redox — CHANGES_REQUESTED** [review]
Some minor feedback, but this is promising!

**Comment by alphonseb** [comment]
This version is published on npm and can be previewed [here](https://algolia-test.zendesk.com/hc/en-gb/requests/new).

**Review by redox — APPROVED** [review]
LGTM

**Pull request merged**